Python/ceval.c

```diff
@@ -4387,6 +4387,15 @@ PyEval_SetTrace(Py_tracefunc func, PyObject *arg)
                            || (tstate->c_profilefunc != NULL));
 }
 
+int
+_PyEval_SetCoroutineOriginTrackingDepth(int new_depth)
+{
+    PyThreadState *tstate = PyThreadState_GET();
```

Comment:
Add `assert(new_depth >= 0)` for cases when somebody uses the C API directly (sys.set_coroutine_origin_tracking_depth has a check for Python users).